test/matrix.py

```diff
@@ -0,0 +1,19 @@
+import translate
+
+
+def matrix(self, code_objects):
+    input_langs = []
+    output_langs = []
+
+    for obj in code_objects:
+        if obj.get_input():
+            input_langs.append(obj)
+        if obj.get_output():
+            output_langs.append(obj)
+
+    for input_lang in input_langs:
+        for output_lang in output_langs:
```

Comment:
Is it intended for a language that is both input and output to get compared to itself?